File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        vec = nums.copy()
        vec.sort()

        left = 0
        right= n-1
        while(left<n):
            if(nums[left]==vec[left]):
                left+=1
            else:
                break

        if(left==n):
            return 0

        while(right>=0):
            if(nums[right]==vec[right]):
                right-=1
            else:
                break

        return right-left+1
        
        

File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int n= nums.size();
        vector<int> vec = nums;
        sort(vec.begin(), vec.end());

        int left = 0;
        int right = n-1;

        if( vec == nums){
            return 0;
        }
        
        while(left<n){
            if(nums[left]!=vec[left]){
                break;
            }
            left++;
        }
        while(right>0){
            if(nums[right]!=vec[right]){
                break;
            }
            right--;
        }
            
        
        return right - left + 1;
        
    }
};